pkg/compose/compose_test: TestComposeCompare failed

pkg/compose/compose_test.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20561503?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20561503?buildTab=artifacts#/) on master @ [ae0c26a5eb2333c0ae3d235b43d4e680db0ae453](https://github.com/cockroachdb/cockroach/commits/ae0c26a5eb2333c0ae3d235b43d4e680db0ae453):


```
        test-1        | === RUN   TestCompare/mutators
        test-1        |     compare_test.go:118: starting test: mutators
        test-1        |     compare_test.go:127: initializing connection: cockroach1
        test-1        |     compare_test.go:127: initializing connection: cockroach2
        test-1        | --- FAIL: TestCompare (2.20s)
        test-1        |     --- FAIL: TestCompare/mutators (1.08s)
        test-1        | panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        test-1        | 	panic: runtime error: invalid memory address or nil pointer dereference
        test-1        | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1bd4d12]
        test-1        | 
        test-1        | goroutine 113 [running]:
        test-1        | testing.tRunner.func1.2({0x2873ea0, 0x4c6b2a0})
        test-1        | 	GOROOT/src/testing/testing.go:1632 +0x230
        test-1        | testing.tRunner.func1()
        test-1        | 	GOROOT/src/testing/testing.go:1635 +0x35e
        test-1        | panic({0x2873ea0?, 0x4c6b2a0?})
        test-1        | 	GOROOT/src/runtime/panic.go:791 +0x132
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*sqlParserImpl).Parse(0xc0012e8cc0, {0x32e0980, 0xc0012e8c58})
        test-1        | 	sql-gen.y:5658 +0x18652
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).parse(0xc0012e8c08, {0xc0002ceb00, 0x575}, {0xc0012e8ff8?, 0x49853d?, 0x573?}, 0x580?, 0x0?)
        test-1        | 	pkg/sql/parser/parse.go:208 +0x172
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).parseWithOptions(0xc0012e8c08, {0xc0002ceb00, 0x576}, {0x4cbc0a0?, 0x1?, 0x0?})
        test-1        | 	pkg/sql/parser/parse.go:177 +0x214
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).ParseWithOptions(...)
        test-1        | 	pkg/sql/parser/parse.go:94
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.ParseWithOptions({0xc0002ceb00, 0x576}, {0x4cbc0a0?, 0xf0?, 0x0?})
        test-1        | 	pkg/sql/parser/parse.go:272 +0x5b
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.Parse(...)
        test-1        | 	pkg/sql/parser/parse.go:265
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.ApplyString(0xc000d246c0, {0xc0002ceb00, 0x576}, {0x5009ee0, 0x0, 0xc001322000?})
        test-1        | 	pkg/sql/randgen/mutator.go:141 +0x70
        test-1        | github.com/cockroachdb/cockroach/pkg/cmd/cmpconn.CompareConns({0x32ffa58?, 0x5009ee0?}, 0x5009ee0?, 0xc0014ffe08, {0x0, 0x0}, {0xc0002ceb00, 0x576}, 0x1)
        test-1        | 	pkg/cmd/cmpconn/conn.go:183 +0x2a6
        test-1        | pkg/compose/compare/compare/compare_test.TestCompare.func2(0xc000ba2340)
        test-1        | 	pkg/compose/compare/compare/compare_test.go:173 +0x6fb
        test-1        | testing.tRunner(0xc000ba2340, 0xc000bfefc0)
        test-1        | 	GOROOT/src/testing/testing.go:1690 +0xf4
        test-1        | created by testing.(*T).Run in goroutine 11
        test-1        | 	GOROOT/src/testing/testing.go:1743 +0x390
        
�[Ktest-1 exited with code 2
        Aborting on container exit...
         Container compare-test-1  Stopping
         Container compare-test-1  Stopped
         Container compare-cockroach2-1  Stopping
         Container compare-cockroach1-1  Stopping
         Container compare-cockroach2-1  Stopped
         Container compare-cockroach1-1  Stopped
        
    compose_test.go:160: exit status 2
--- FAIL: TestComposeCompare (10.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #153083 pkg/compose/compose_test: TestComposeCompare failed [connection closed] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.1.24-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/compose/compose_test.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20562554?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20562554?buildTab=artifacts#/) on master @ [f520554bbd8f278a97a0bef459f7f671c01f4564](https://github.com/cockroachdb/cockroach/commits/f520554bbd8f278a97a0bef459f7f671c01f4564):


```
        test-1        | === RUN   TestCompare/mutators
        test-1        |     compare_test.go:118: starting test: mutators
        test-1        |     compare_test.go:127: initializing connection: cockroach1
        test-1        |     compare_test.go:127: initializing connection: cockroach2
        test-1        | --- FAIL: TestCompare (2.00s)
        test-1        |     --- FAIL: TestCompare/mutators (0.92s)
        test-1        | panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        test-1        | 	panic: runtime error: invalid memory address or nil pointer dereference
        test-1        | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1bd4d12]
        test-1        | 
        test-1        | goroutine 114 [running]:
        test-1        | testing.tRunner.func1.2({0x2873ea0, 0x4c6b2a0})
        test-1        | 	GOROOT/src/testing/testing.go:1632 +0x230
        test-1        | testing.tRunner.func1()
        test-1        | 	GOROOT/src/testing/testing.go:1635 +0x35e
        test-1        | panic({0x2873ea0?, 0x4c6b2a0?})
        test-1        | 	GOROOT/src/runtime/panic.go:791 +0x132
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*sqlParserImpl).Parse(0xc0022f46c0, {0x32e0980, 0xc0022f4658})
        test-1        | 	sql-gen.y:5658 +0x18652
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).parse(0xc0022f4608, {0xc002468a00, 0x147a}, {0xc0022f49f8?, 0x7f52ef5ee650?, 0x146f?}, 0x1500?, 0x0?)
        test-1        | 	pkg/sql/parser/parse.go:208 +0x172
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).parseWithOptions(0xc0022f4608, {0xc002468a00, 0x147b}, {0x4cbc0a0?, 0x1?, 0x0?})
        test-1        | 	pkg/sql/parser/parse.go:177 +0x214
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).ParseWithOptions(...)
        test-1        | 	pkg/sql/parser/parse.go:94
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.ParseWithOptions({0xc002468a00, 0x147b}, {0x4cbc0a0?, 0x30?, 0x0?})
        test-1        | 	pkg/sql/parser/parse.go:272 +0x5b
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.Parse(...)
        test-1        | 	pkg/sql/parser/parse.go:265
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.ApplyString(0xc000b5e120, {0xc002468a00, 0x147b}, {0x5009ee0, 0x0, 0x49853d?})
        test-1        | 	pkg/sql/randgen/mutator.go:141 +0x70
        test-1        | github.com/cockroachdb/cockroach/pkg/cmd/cmpconn.CompareConns({0x32ffa58?, 0x5009ee0?}, 0x5009ee0?, 0xc0020bbe08, {0x0, 0x0}, {0xc002468a00, 0x147b}, 0x1)
        test-1        | 	pkg/cmd/cmpconn/conn.go:183 +0x2a6
        test-1        | pkg/compose/compare/compare/compare_test.TestCompare.func2(0xc000714000)
        test-1        | 	pkg/compose/compare/compare/compare_test.go:173 +0x6fb
        test-1        | testing.tRunner(0xc000714000, 0xc000b56360)
        test-1        | 	GOROOT/src/testing/testing.go:1690 +0xf4
        test-1        | created by testing.(*T).Run in goroutine 10
        test-1        | 	GOROOT/src/testing/testing.go:1743 +0x390
        
�[Ktest-1 exited with code 2
        Aborting on container exit...
         Container compare-test-1  Stopping
         Container compare-test-1  Stopped
         Container compare-cockroach2-1  Stopping
         Container compare-cockroach1-1  Stopping
         Container compare-cockroach2-1  Stopped
         Container compare-cockroach1-1  Stopped
        
    compose_test.go:160: exit status 2
--- FAIL: TestComposeCompare (9.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #153083 pkg/compose/compose_test: TestComposeCompare failed [connection closed] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.1.24-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/compose/compose_test.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20563524?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20563524?buildTab=artifacts#/) on master @ [14ff833383127fa428b913c5281a8a6c862e2a41](https://github.com/cockroachdb/cockroach/commits/14ff833383127fa428b913c5281a8a6c862e2a41):


```
        test-1        | === RUN   TestCompare/mutators
        test-1        |     compare_test.go:118: starting test: mutators
        test-1        |     compare_test.go:127: initializing connection: cockroach1
        test-1        |     compare_test.go:127: initializing connection: cockroach2
        test-1        | --- FAIL: TestCompare (2.20s)
        test-1        |     --- FAIL: TestCompare/mutators (1.09s)
        test-1        | panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        test-1        | 	panic: runtime error: invalid memory address or nil pointer dereference
        test-1        | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1bd4d12]
        test-1        | 
        test-1        | goroutine 94 [running]:
        test-1        | testing.tRunner.func1.2({0x2873ea0, 0x4c6b2a0})
        test-1        | 	GOROOT/src/testing/testing.go:1632 +0x230
        test-1        | testing.tRunner.func1()
        test-1        | 	GOROOT/src/testing/testing.go:1635 +0x35e
        test-1        | panic({0x2873ea0?, 0x4c6b2a0?})
        test-1        | 	GOROOT/src/runtime/panic.go:791 +0x132
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*sqlParserImpl).Parse(0xc00057d8c0, {0x32e0980, 0xc00057d858})
        test-1        | 	sql-gen.y:5658 +0x18652
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).parse(0xc00057d808, {0xc002374000, 0x4f855}, {0xc00057dbf8?, 0x7f2267e3fe88?, 0x0?}, 0x50000?, 0x0?)
        test-1        | 	pkg/sql/parser/parse.go:208 +0x172
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).parseWithOptions(0xc00057d808, {0xc002374000, 0x4f856}, {0x4cbc0a0?, 0x1?, 0x0?})
        test-1        | 	pkg/sql/parser/parse.go:177 +0x214
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.(*Parser).ParseWithOptions(...)
        test-1        | 	pkg/sql/parser/parse.go:94
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.ParseWithOptions({0xc002374000, 0x4f856}, {0x4cbc0a0?, 0xe0?, 0x0?})
        test-1        | 	pkg/sql/parser/parse.go:272 +0x5b
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/parser.Parse(...)
        test-1        | 	pkg/sql/parser/parse.go:265
        test-1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.ApplyString(0xc000d41170, {0xc002374000, 0x4f856}, {0x5009ee0, 0x0, 0xc000b93001?})
        test-1        | 	pkg/sql/randgen/mutator.go:141 +0x70
        test-1        | github.com/cockroachdb/cockroach/pkg/cmd/cmpconn.CompareConns({0x32ffa58?, 0x5009ee0?}, 0x5009ee0?, 0xc001d9fe08, {0x0, 0x0}, {0xc002374000, 0x4f856}, 0x1)
        test-1        | 	pkg/cmd/cmpconn/conn.go:183 +0x2a6
        test-1        | pkg/compose/compare/compare/compare_test.TestCompare.func2(0xc000d5e000)
        test-1        | 	pkg/compose/compare/compare/compare_test.go:173 +0x6fb
        test-1        | testing.tRunner(0xc000d5e000, 0xc000d366c0)
        test-1        | 	GOROOT/src/testing/testing.go:1690 +0xf4
        test-1        | created by testing.(*T).Run in goroutine 33
        test-1        | 	GOROOT/src/testing/testing.go:1743 +0x390
        
�[Ktest-1 exited with code 2
        Aborting on container exit...
         Container compare-test-1  Stopping
         Container compare-test-1  Stopped
         Container compare-cockroach1-1  Stopping
         Container compare-cockroach2-1  Stopping
         Container compare-cockroach2-1  Stopped
         Container compare-cockroach1-1  Stopped
        
    compose_test.go:160: exit status 2
--- FAIL: TestComposeCompare (9.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #153083 pkg/compose/compose_test: TestComposeCompare failed [connection closed] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.1.24-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/randgen/BUILD.bazel
+++ b/pkg/sql/randgen/BUILD.bazel
@@ -30,6 +30,7 @@ go_library(
         "//pkg/sql/oidext",
         "//pkg/sql/parser",
         "//pkg/sql/pgrepl/lsn",
+        "//pkg/sql/plpgsql/parser",
         "//pkg/sql/rowenc",
         "//pkg/sql/rowenc/valueside",
         "//pkg/sql/sem/cast",

--- a/pkg/sql/randgen/mutator.go
+++ b/pkg/sql/randgen/mutator.go
@@ -19,6 +19,9 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/parser"
+	// Ensure that sql/parser.ParseDoBlockFn is injected from the PLpgSQL
+	// parser.
+	_ "github.com/cockroachdb/cockroach/pkg/sql/plpgsql/parser"
 	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/idxtype"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"

